Completed homework 11 sprint

RURURuina opened the pull request:
S BOGOM

RURURuina committed:
Completed homework 11 sprint
S BOGOM

IliaAnK commented on app/src/main/java/com/practicum/playlistmaker/SearchActivity.kt:
```diff
@@ -82,6 +100,45 @@ class SearchActivity : AppCompatActivity() {
     }
 
+    private fun performSearch (query: String) {
+        val retrofit = Retrofit.Builder()
```
Рекомендация:
Обычно, на всё приложение существует лишь несколько экземпляров `Retrofit` клиента. Каждый из которых представлен является синглтон объектом. 
Но сейчас, каждый раз, когда выполняется новый поиск создаётся новый инстанс `Retrofit` клиента.

`Retrofit` достаточно "тяжёлый" объект и, как минимум, его создание это дорогостоящая операция + в памяти остаются "висеть" его не используемые инстансы, а это может приводить к частой очистки памяти Garbage коллетором. Каждый раз, когда Garbage коллетор начинает свою работу, он останавливает выполнение программы, следовательно надо стараться триггерить его как можно реже.

На данном этапе, я могу порекомендовать вынести создание `Retrofit` клиента и `service` на уровень класса. И добавить ленивую инициализацию с помощью делегата `by lazy`.

RURURuina committed:
Completed homework 11 sprint
S BOGOM
Вывел клиент Retrofit в отдельный класс и инициализировал в SearchActivity используя by lazy
